### PR TITLE
safe client: Show safe creation in progress and error states

### PR DESCRIPTION
**Pull request by jurgenwerk**
This PR aims to improve the user experience when creating a safe. It tells the user when creation and indexing is in progress, and shows error states if errors happen while getting the gas cost, checking the balance, provisioning or indexing.

Opening the modal will start checking the gas cost and compare it against users' balance:
image

When the balance is insufficient:
image

When getting the gas cost or fetching balance fail (these should ideally be separated but good for now IMO):
image

When balance is sufficient:
image

When creating the safe:
telegram-cloud-photo-size-4-5850565774886943958-y

When indexing (waiting for the safe to appear in our hosted subgraph):
image

When indexing is done and we load the safe so that it appears in the sidebar:
image

When an error happens during safe creation:
image

When an error happens during waiting for the safe to be indexed (polling fails or timeout of 2 mins is reached): 
image

### CODE REVIEW

**Review by lukemelia — APPROVED** [review]
Nice job isolating these states. Some product thoughts:

1) In addition to showing the estimated gas cost in the native gas coin, it would be nice to show the USD equivalent.
2) It is non-obvious why a user should have to wait for "Indexing..." after "Provisioning...". Perhaps "Confirming..." would be more suitable?

File: packages/safe-tools-client/app/components/create-safe/index.gts [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+    let safeAddress: string = '';
+
+
+    try {
+      ({ safeAddress } = await taskFor(this.createSafe).perform());
+    } catch (e) {
+      console.log(e) // TODO: Sentry
+      return;
+    }
+    this.safeCreated = true;
```
[the remark this answers is not in the document]
Comment (jurgenwerk) [reply packages/safe-tools-client/app/components/create-safe/index.gts]: Great idea, added.

File: packages/safe-tools-client/app/components/create-safe-modal/index.gts [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+          {{else}}
+            {{#if @comparingBalanceToGasCostErrorMessage}}
+              <div class='safe-setup-modal__section-wallet-info'>
+                {{svgJar
+                  'icon-x-circle-ht'
+                  class='safe-setup-modal__section-icon'
+                  style=(cssVar
+                    icon-color='var(--boxel-red)'
+                  )
+                }}
```
[the remark this answers is not in the document]
Comment (jurgenwerk) [reply packages/safe-tools-client/app/components/create-safe-modal/index.gts]: Agree, added.

File: packages/safe-tools-client/app/components/create-safe/index.gts [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+
+  @tracked isModalOpen = false;
+
+  @tracked isLoadingGasInfo = false;
+  @tracked hasEnoughBalance = false;
+  @tracked gasCostDisplay =  '';
+  @tracked safeCreated =  false;
+  @tracked comparingBalanceToGasCostErrorMessage = '';
+
+  @action async openCreateSafeModal() {
```
[the remark this answers is not in the document]
Comment (jurgenwerk) [reply packages/safe-tools-client/app/components/create-safe/index.gts]: Adjusted.

**Review by lukemelia — APPROVED** [review]

**Comment by kdowlin** [comment]
> Nice job isolating these states. Some product thoughts:
> 
> 1. In addition to showing the estimated gas cost in the native gas coin, it would be nice to show the USD equivalent.
> 2. It is non-obvious why a user should have to wait for "Indexing..." after "Provisioning...". Perhaps "Confirming..." would be more suitable?

Agree - also the CTA of Provision is a bit odd.
What if you did:
CTA “Create”
Processing word: “Creating…” and add help text “This may take up to xx mins. During this time your payment safe is created and indexed.” Or something like that. @jurgenwerk @lukemelia

**Pull request merged**